cabal-install/Distribution/Client/CmdWriteAutogenFiles.hs

```diff
@@ -0,0 +1,220 @@
+-- | cabal-install CLI command: build
```

Comment:
I think write-autogen-files should be a `new-build` flag instead. The overhead of having a separate command is just not called for when all we need to do is make new-build quit early.

Reply:
What would be a nice flag?
E.g. `cabal new-build --write-autogen-files`?
That should not really compile the whole project, right?

Reply:
I agree with @DanielG -- let's avoid the extra superflous cost of duplicating command structures and have it be a flag of `v2-build`.

In general, `v2-build` should have some flags to stop after a specific phase of the build. Stopping right after having generated the autogen files would be one such phase.  Or stopping right after the `configure` phase would be another abortion-point.

Assuming there is a linear dependency chain, it might make sense to design this as an enum field, i.e.

```
--stop-after-buildphase={configure,autogen,...}
```

Reply:
For some prior art (which made me think of this), `rpmbuild` has the following flags:

```
Usage: rpmbuild [OPTION...]

Build options with [ <specfile> | <tarball> | <source package> ]:
      -bp                        build through %prep (unpack sources and apply patches) from <specfile>
      -bc                        build through %build (%prep, then compile) from <specfile>
      -bi                        build through %install (%prep, %build, then install) from <specfile>
      -bl                        verify %files section from <specfile>
      -ba                        build source and binary packages from <specfile>
      -bb                        build binary package only from <specfile>
      -bs                        build source package only from <specfile>
      -tp                        build through %prep (unpack sources and apply patches) from <tarball>
      -tc                        build through %build (%prep, then compile) from <tarball>
      -ti                        build through %install (%prep, %build, then install) from <tarball>
      -ta                        build source and binary packages from <tarball>
      -tb                        build binary package only from <tarball>
      -ts                        build source package only from <tarball>
      --rebuild                  build binary package from <source package>
      --recompile                build through %install (%prep, %build, then install) from <source package>
      --buildroot=DIRECTORY      override build root
      --clean                    remove build tree when done
      --nobuild                  do not execute any stages of the build
      --nodeps                   do not verify build dependencies
      --nodirtokens              generate package header(s) compatible with (legacy) rpm v3 packaging
      --noclean                  do not execute %clean stage of the build
      --nocheck                  do not execute %check stage of the build
      --rmsource                 remove sources when done
      --rmspec                   remove specfile when done
      --short-circuit            skip straight to specified stage (only for c,i)
      --target=CPU-VENDOR-OS     override target platform
```

Note all the short two-letter  flags for controlling which build-phases to execute...

Reply:
Resolve for now, different pull request, eventually.